### PR TITLE
feature/#569 added warning for the use of Scope.PIPELINE

### PR DESCRIPTION
**Pull request by toan-quach**
#569

### CODE REVIEW

**Review by trgiangdo — COMMENTED** [review]
Because the default Scope now is PIPELINE, will this show too many warnings?

I think we can only show the warning after changing the default scope.

**Comment by toan-quach** [comment]
@trgiangdo 
yes, you're absolutely right that there will be a lot of messages. I put it in the from_model function simply because that's the only "guaranteed" way the message will pop up. We can also put it when accessing the attribute "scope" of dn or task. But the message will only appear when the user tries to access the "scope" attribute.
As for the default Scope value, I agree, we should change it.

@jrobinAV
Do you think this ticket is also the right place to change the default scope value (Imho, it's a yes)? And if yes, what would be the right default value (I'll say Scope.SCENARIO)?

**Comment by jrobinAV** [comment]
> Scope.SCENARIO

The Scope.SCENARIO is already the default value.

File: src/taipy/core/data/_data_repository.py [hunk 1]
```diff
@@ -205,6 +208,10 @@ def _from_model(self, model: _DataNodeModel):
         validity_period = None
         if model.validity_seconds is not None and model.validity_days is not None:
             validity_period = timedelta(days=model.validity_days, seconds=model.validity_seconds)
+
+        if model.scope == Scope.PIPELINE:
+            _warn_deprecated(Scope.PIPELINE, Scope.SCENARIO)
```
Comment (jrobinAV) [inline src/taipy/core/data/_data_repository.py]: WHy do you use Scope enum object here, and string in data_node warning?

**Review by jrobinAV — DISMISSED** [review]

File: src/taipy/core/data/_data_repository.py (hunk 1 above)
Comment (toan-quach) [reply src/taipy/core/data/_data_repository.py]: Oh you're right, I noticed the deprecated value is expecting a str, so I keep it as str in data_node. But you're right, we can simply pass the scope value instead

Comment (toan-quach) [reply src/taipy/core/data/_data_repository.py]: same for in task.py

Comment (toan-quach) [reply src/taipy/core/data/_data_repository.py]: I have changed it to ```_warn_deprecated(f"`{Scope.PIPELINE}`", suggest="other `Scope` value")``` to be consistent with the config checker message. Let me know if that makes sense to you

**Review by trgiangdo — COMMENTED** [review]
LGTM! Let's wait for the test to pass and I believe this is ready to merge.

**Review by trgiangdo — APPROVED** [review]

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**